src/reactive-graphql.ts

```diff
@@ -0,0 +1,400 @@
+import { Observable, of, from, throwError, isObservable } from "rxjs";
+import { concatMap, map, combineLatest } from "rxjs/operators";
+
+import {
+  DefinitionNode,
+  DocumentNode,
+  getNamedType,
+  GraphQLInterfaceType,
+  GraphQLObjectType,
+  GraphQLSchema,
+  GraphQLType,
+  SelectionNode,
+  FieldNode,
+  GraphQLField,
+  GraphQLFieldResolver,
+  isTypeSystemDefinitionNode,
+  isTypeSystemExtensionNode,
+  Kind,
+  ArgumentNode,
+  isScalarType,
+  isEnumType,
+  isObjectType,
+  parse
+} from "graphql";
+
+// WARNING: This is NOT a spec complete graphql implementation
+// https://facebook.github.io/graphql/October2016/
+
+interface TypeMap {
+  [key: string]: GraphQLType;
+}
+
+interface Schema extends GraphQLSchema {
+  _typeMap?: TypeMap;
+}
+
+interface OperationNode {
+  operation: "query" | "mutation";
+}
+
+type SchemaNode = SelectionNode | DefinitionNode;
+
+function isOperationDefinition(node: any): node is OperationNode {
+  return node.kind === Kind.OPERATION_DEFINITION;
+}
+function isFieldNode(node: SchemaNode): node is FieldNode {
+  return node.kind === Kind.FIELD;
+}
+
+// We don't treat OperationDefinitions as Definitions but as entry points for our execution
+function isDefinitionNode(node: SchemaNode): node is DefinitionNode {
+  return (
+    node.kind === Kind.FRAGMENT_DEFINITION ||
+    isTypeSystemDefinitionNode(node) ||
+    isTypeSystemExtensionNode(node)
+  );
+}
+
+interface FieldWithResolver extends GraphQLField<any, any> {
+  resolve: GraphQLFieldResolver<any, any, any>;
+}
+
+function isFieldWithResolver(
+  field: GraphQLField<any, any>
+): field is FieldWithResolver {
+  return field.resolve instanceof Function;
+}
+
+export default function graphql<T = object>(
+  schema: Schema,
+  query: string | DocumentNode,
+  rootValue?: any,
+  context: object = {},
+  variables: object = {}
+): Observable<{ data?: T; errors?: string[] }> {
```

Comment:
```
Observable<ExecutionResult<TData>>

// witt:
export interface ExecutionResult<TData = ExecutionResultDataDefault> {
    errors?: ReadonlyArray<GraphQLError>;
    data?: TData;
}
```
Note the difference of type on the errors key, and indeed note that line 82, [`parse` throwing GraphQLErrors](https://github.com/graphql/graphql-js/blob/master/src/language/parser.js#L123), GraphQLErrors are pushed in errors.

Reply:
Yeah that might have been a mistake, normally we throw the observable, we should do this here, too. I mean not being able to parse is a prime example for something that a client can not live without in contrast to errors in field resolvers where one could argue for partial delivery of the data tree.

Reply:
Given GQL spec, [any request validation error _should be reported in the list of “errors” in the response and the request must fail without execution_](https://facebook.github.io/graphql/June2018/#sec-Validating-Requests).
I'm not sure if an error in the syntax should be understood as "validation error" and I'm not sure what "request fail" mean.
 
Anyway, in the reference implementation, [they decided to add any SyntaxError to the list of errors](https://github.com/graphql/graphql-js/blob/master/src/graphql.js#L203) and to resolve normallt the returned.

Given this, while I understand your POV, I think that the current implementation is correct and I'd let it as it. 🤷‍♂️

Reply:
ok now I realize that every error is actually thrown away with the current implementation, so by consistency this one should the same. However, I think it would be more correct to aggregate errors in the `errors` array in the returned results => https://github.com/mesosphere/reactive-graphql/issues/13